client/starwhale/api/_impl/dataset/loader.py

```diff
@@ -15,31 +13,37 @@
 from starwhale.base.uri import URI
 from starwhale.base.type import URIType, InstanceType, DataFormatType, ObjectStoreType
 from starwhale.core.dataset.type import BaseArtifact
-from starwhale.core.dataset.model import Dataset
 from starwhale.core.dataset.store import FileLikeObj, ObjectStore, DatasetStorage
-from starwhale.core.dataset.tabular import TabularDataset, TabularDatasetRow
+from starwhale.core.dataset.tabular import (
+    TabularDataset,
+    TabularDatasetRow,
+    TabularDatasetSessionConsumption,
+)
 
 
 class DataLoader(metaclass=ABCMeta):
     def __init__(
         self,
         dataset_uri: URI,
-        start: int = 0,
-        end: int = sys.maxsize,
-        logger: t.Union[loguru.Logger, None] = None,
+        start: t.Optional[t.Any] = None,
+        end: t.Optional[t.Any] = None,
+        logger: t.Optional[loguru.Logger] = None,
+        session_consumption: t.Optional[TabularDatasetSessionConsumption] = None,
     ):
         self.dataset_uri = dataset_uri
+        self.logger = logger or _logger
         self.start = start
         self.end = end
-        self.logger = logger or _logger
+
         # TODO: refactor TabularDataset with dataset_uri
         # TODO: refactor dataset, tabular_dataset and standalone dataset module
         self.tabular_dataset = TabularDataset.from_uri(
             dataset_uri, start=start, end=end
         )
+        self.session_consumption = session_consumption
```

Comment:
Add range checking for loader and `session_consumption`?
Different ranges will cause unexpected behavior.

Reply:
ok

Reply:
done